### PR TITLE
[objc] Add array ref parameter support. Fixes Issue #314

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: objcgen/objcgenerator.cs [hunk 1]
```diff
@@ -509,24 +509,129 @@ void Generate (ParameterInfo [] parameters, bool isExtension, out string postInv
...
 		}
 
-		void GenerateArgumentArray (string parameterName, string argumentName, Type t, ref StringBuilder post)
+		string GenerateArgumentArrayPost (string parameterName, string argumentName, Type t)
+		{
+			var postwriter = new SourceWriter {
+				Indent = 1
+			};
+			var typecode = Type.GetTypeCode (t);
+			var parrlenght = $"__p{parameterName}length";
```
Comment (rolfbjarne) [inline objcgen/objcgenerator.cs]: Typo: `parrlenght` instead of `parrlength`.

File: objcgen/objcgenerator.cs [hunk 2]
```diff
@@ -509,24 +509,129 @@ void Generate (ParameterInfo [] parameters, bool isExtension, out string postInv
...
+			else {
+				postwriter.WriteLine ($"{pbuff}[{pindex}] = {presobj};");
+				postwriter.Indent--;
+				postwriter.WriteLine ("}");
+				postwriter.WriteLine ($"*{parameterName} = [[NSArray alloc] initWithObjects: {pbuff} count: {parrlenght}];");
+				postwriter.WriteLine ($"for ({pindex} = 0; {pindex} < {parrlenght}; {pindex}++)");
+				postwriter.Indent++;
+				postwriter.WriteLine ($"{pbuff} [{pindex}] = nil;");
+				postwriter.Indent--;
+				postwriter.WriteLine ($"free({pbuff});");
```
Comment (rolfbjarne) [inline objcgen/objcgenerator.cs]: Style: space between function name and opening parenthesis.

**Comment by dalexsoto** [comment]
@rolfbjarne fixed.

**Comment by dalexsoto** [comment]
Will add NSDecimal/decimal ref in a separate PR to unblock Tim ASAP

**Pull request merged**